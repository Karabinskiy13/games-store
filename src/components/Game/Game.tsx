import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import { Games } from '../../../types';
import { Button, GameItem } from '../../styles';

const Game = ({ game, showModal }: Games) => {
  const posterLink = `http://${game.fields.poster?.fields.file.url}`;
  const showDescription = `/games/${game.fields.name}`;

  return (
    <GameItem>
      <Image
        className="poster"
        src={posterLink}
        width={270}
        height={400}
        alt="poster"
        data-testId="poster"
        onClick={() => showModal(true)}
      />
      <div className="name">{game.fields.name}</div>
      <div data-testId="price">{game.fields.price}UAH</div>
      <div data-testId="releaseDate">Release Date: {game.fields.releaseDate}</div>
      <Link href={showDescription}>
        <Button data-testId="button" className="button">
          {game.fields.showFull}
        </Button>
      </Link>
    </GameItem>
  );
};

export default Game;
