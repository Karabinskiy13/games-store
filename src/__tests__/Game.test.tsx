import React from 'react';
import { jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IGames, IGamesFields } from '../../contentful';
import Game from '../components/Game/Game';

const singleGame: IGames = {
  id: 1,
  quantity: 10,
  prevState: null,
  fields: {
    name: 'name',
    description: 'description',
    price: 100,
    releaseDate: '2020',
    platform: 'PS4',
    genres: 'action'
  },
  sys: {
    id: '1',
    type: 'type',
    createdAt: '01.01.2023',
    updatedAt: '02.02.2023',
    locale: 'locale',
    contentType: {
      sys: {
        id: 'games',
        linkType: 'ContentType',
        type: 'Link'
      }
    }
  }
};

describe('<SinglePicture>', () => {
  test('Should render component', () => {
    const { asFragment } = render(<Game game={singleGame} showModal={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should call a callback when the card is clicked', async () => {
    const showModalCb = jest.fn(() => true);
    render(<Game game={singleGame} showModal={showModalCb} />);
    const user = userEvent.setup();
    await user.click(screen.getByAltText('poster'));
    expect(showModalCb).toBeCalledWith(true);
  });

  test('Should display image from props', async () => {
    render(<Game game={singleGame} showModal={jest.fn()} />);
    await screen.findAllByAltText('poster');
    expect(screen.getByAltText('poster')).toHaveAttribute('src', '/image.png');
  });
});
