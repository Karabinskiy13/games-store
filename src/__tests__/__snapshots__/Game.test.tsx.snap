// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePicture> Should render component 1`] = `
<DocumentFragment>
  <div
    class="Games__GameItem-sc-itkhej-2 fhPXZl"
  >
    <img
      alt="poster"
      class="poster"
      data-nimg="1"
      data-testid="poster"
      decoding="async"
      height="400"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fundefined&w=640&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fundefined&w=384&q=75 1x, /_next/image?url=http%3A%2F%2Fundefined&w=640&q=75 2x"
      style="color: transparent;"
      width="270"
    />
    <div
      class="name"
    >
      name
    </div>
    <div
      data-testid="price"
    >
      100UAH
    </div>
    <div>
      Release Date: 2020
    </div>
    <button
      class="Games__Button-sc-itkhej-1 kRtork button"
    />
  </div>
</DocumentFragment>
`;
